lib/badge/runner.rb

```diff
@@ -126,15 +120,11 @@ def add_shield(icon, result, shield, alpha_channel, shield_gravity, shield_no_re
 
     def load_shield(shield_string)
       url = Badge.shield_base_url + Badge.shield_path + shield_string + (@@rsvg_enabled ? ".svg" : ".png")
-      file_name = shield_string + (@@rsvg_enabled ? ".svg" : ".png")
 
       UI.verbose "Trying to load image from shields.io. Timeout: #{Badge.shield_io_timeout}s".blue
       UI.verbose "URL: #{url}".blue
 
-      Curl::Easy.download(url, file_name)
-      MiniMagick::Image.open(file_name) unless @@rsvg_enabled
-
-      File.open(file_name)
+      MiniMagick::Image.open(url)
```

Comment:
We can open images via URL with MiniMagick.

https://github.com/minimagick/minimagick#usage